doc/specs/thread_tagging.md

```diff
@@ -0,0 +1,215 @@
+
+* The user has permission to view the forum.
+
+**Trigger:** The user is viewing the forum.
+
+**Postconditions:** N/A
+
+**Basic Flow:**
+
+1. The user navigates to the forum.
+
+**Non-Functional Requirements:**
+
+* Thread labels must be visible somewhere on the forum overview.
+* Thread labels must not be included in thread titles using brackets, or any other method.
+* Thread tags must be visible somewhere on the forum overview, if they can be added
+  in a visually appealing manner.
+
+### 1.1 Viewing a thread in a tagging-enabled forum
+
+**Primary Actor:** A user
+
+**Brief:** When viewing a thread in a tagging-enabled forum, the tags and labels must be
+clearly visible.
+
+**Preconditions:** N/A
+
+**Trigger:** The user opens the thread.
+
+**Postconditions:** N/A
+
+**Basic Flow:**
+
+1. The user opens the thread.
+
+**Non-Functional Requirements:**
+
+* Thread labels and tags must be visible.
+
+### 2.0 Creating a thread in a tagging-enabled forum
+
+**Primary Actor:** A user
+
+**Brief:** When a user creates a thread in a tagging-enabled forum, they must be presented
+with a choice of a label and tags.
+
+**Preconditions:** N/A
+
+**Trigger:** When the user is creating a new thread.
+
+**Postconditions:**
+
+* The thread is created, with the appropriate labels and tags added to it.
+
+**Basic Flow:**
+
+1. The user selects to create a new thread
+2. The user is presented with the thread creation template, along with an option for a label.
+3. When the user selects a label, they are presented with options for tags.
+4. The user submits the form, and the thread is created.
+
+**Applicable business rules:**
+
+* A label must be selected before tags.
+
+### 3.0 Editing tags on a thread
+
+**Primary Actor:** A user
+
+**Brief:** Tagged threads must be able to be edited to ensure accuracy and relevancy.
+
+**Preconditions:**
+
+* One of the following three:
+  * The user is the creator of the thread and the thread is not tagged locked.
+  * The user has the tag management permission enabled and the thread is not tagged locked.
+  * The user belongs to a user class that has been granted tag editing permission in the specified
+    forum, and the thread is not tag locked.
+  * The user has the tag administration permission enabled.
+
+**Trigger:** The user chooses to edit the tags on a thread.
+
+**Postconditions:**
+
+* The updated labels and tags are saved.
+
+**Basic Flow:**
+
+1. The user selects the option to edit the thread's tags.
+2. The user is presented with a tag manager and makes the desired changes.
+3. The user saves the changes.
+
+**Functional Requirements:**
+
+* When a thread's labels/tags are modified after the thread creation, the changes must be documented
+  in the thread notes section.
+
+### 4.0 Tag locking a thread
+
+**Primary Actor:** A user with the tag administration permission enabled
+
+**Brief:** Threads must be able to be thread locked to prevent any further editing.
+
+**Preconditions:** N/A
```

Comment:
I mentioned the permission in the primary actor section, but I moved it to here.

Resolved.